test/tools/Modules/HelpersCommon/HelpersCommon.psm1

```diff
@@ -11,14 +11,35 @@ function Wait-UntilTrue
 
     # Loop until the script block evaluates to true
     while (-not ($sb.Invoke())) {
+        # If the timeout period has passed, return false
```

Comment:
Interesting, make sense to enhance `Invoke()` to support timeout?

Reply:
You can do async invocation via powershell API, but the experience is not optimal.